core/src/main/scala/akka/persistence/jdbc/state/scaladsl/DurableStateSequenceActor.scala

```diff
@@ -138,24 +188,24 @@ class DurableStateSequenceActor[A](stateStore: JdbcDurableStateStore[A], config:
 
     val (nextMax, _, missingElems) =
       // using the ordering elements that were fetched, we verify if there are any gaps
-      elements.foldLeft[(OrderingId, OrderingId, MissingElements)](
-        (currentMaxOrdering, currentMaxOrdering, MissingElements.empty)) {
-        case ((currentMax, previousElement, missing), currentElement) =>
+      elements.foldLeft[(GlobalOffset, GlobalOffset, MissingElements)](
+        (currentMaxOffset, currentMaxOffset, MissingElements.empty)) {
+        case ((currentMax, previousOffset, missing), currentElement) =>
           // we must decide if we move the cursor forward
           val newMax =
-            if ((currentMax + 1).until(currentElement).forall(givenUp.contains)) {
+            if ((currentMax + 1).until(currentElement.offset).forall(givenUp.contains)) {
```

Comment:
@patriknw do we have to move the cursor one element at a time. In case of journal impl we used to store all the offsets in sequence - here we store the last offset only. Hence the strategy that I implemented was :

```
for all elements in the results
    if (total revision changes == total offset movement), then we have no gaps
```

This also takes into consideration the fact that we are using the READ COMMITTED isolation level and hence we cannot see upsert for revision `n+1` without seeing the upsert for revision `n`